### ISSUE TITLE
Incompatibility with Darklaf surfaces when setting a window icon

### ISSUE THREAD

**Issue by Syer10** [JetBrains/compose-jb#1097]
Adding a issue in response to slack.
When using the Swing Laf library called DarkLaf, if you set a window icon for your application, the window will never actually appear and it will throw an exception. 
```java
15:24:18.520 [AWT-EventQueue-0 @coroutine#23] ERROR/UncaughtException: Uncaught exception in thread [AWT-EventQueue-0 @coroutine#23@40]
java.lang.UnsupportedOperationException: getSource() not supported
	at androidx.compose.ui.graphics.PainterImage.getSource(DesktopImageConverters.desktop.kt:137) ~[ui-graphics-desktop-1.0.0-alpha4-build315.jar:?]
	at java.awt.Image.getScaledInstance(Image.java:177) ~[?:?]
	at com.github.weisj.darklaf.platform.windows.ui.WindowsTitlePane.updateSystemIcon(WindowsTitlePane.java:657) ~[darklaf-windows-2.7.2.jar:2.7.2]
	at com.github.weisj.darklaf.platform.windows.ui.WindowsTitlePane.install(WindowsTitlePane.java:260) ~[darklaf-windows-2.7.2.jar:2.7.2]
	at com.github.weisj.darklaf.platform.decorations.CustomTitlePane.addNotify(CustomTitlePane.java:40) ~[darklaf-platform-base-2.7.2.jar:2.7.2]
	at java.awt.Container.addNotify(Container.java:2800) ~[?:?]
	at javax.swing.JComponent.addNotify(JComponent.java:4783) ~[?:?]
	at java.awt.Container.addNotify(Container.java:2800) ~[?:?]
	at javax.swing.JComponent.addNotify(JComponent.java:4783) ~[?:?]
	at javax.swing.JRootPane.addNotify(JRootPane.java:729) ~[?:?]
	at java.awt.Container.addNotify(Container.java:2800) ~[?:?]
	at java.awt.Window.addNotify(Window.java:783) ~[?:?]
	at java.awt.Frame.addNotify(Frame.java:490) ~[?:?]
	at java.awt.Window.show(Window.java:1045) ~[?:?]
	at java.awt.Component.show(Component.java:1717) ~[?:?]
	at java.awt.Component.setVisible(Component.java:1664) ~[?:?]
	at java.awt.Window.setVisible(Window.java:1028) ~[?:?]
	at androidx.compose.ui.window.AwtWindow_desktopKt$AwtWindow$4$1.invokeSuspend(AwtWindow.desktop.kt:122) ~[ui-desktop-1.0.0-alpha4-build315.jar:?]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) ~[kotlin-stdlib-1.5.21.jar:1.5.21-release-314 (1.5.21)]
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106) ~[kotlinx-coroutines-core-jvm-1.5.1.jar:?]
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:316) ~[?:?]
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770) ~[?:?]
	at java.awt.EventQueue$4.run(EventQueue.java:721) ~[?:?]
	at java.awt.EventQueue$4.run(EventQueue.java:715) ~[?:?]
	at java.security.AccessController.doPrivileged(AccessController.java:391) [?:?]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85) [?:?]
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:740) [?:?]
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203) [?:?]
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124) [?:?]
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113) [?:?]
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109) [?:?]
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) [?:?]
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:90) [?:?]
```

**Comment by igordmn** [comment]
The same crash happens when we use icon with enabled = false (reported in [Slack](https://kotlinlang.slack.com/archives/C01D6HTPATV/p1631808744286100)):
```
MenuBar {
    Menu("File", mnemonic = 'F') {
        Item(
            "New",
            icon = rememberVectorPainter(Icons.Rounded.NoteAdd),
            onClick = { model.newFile() },
            shortcut = KeyShortcut(Key.N, ctrl = true),
            enabled = false
        )
    }
}
```
crash:
```
Exception in thread "AWT-EventQueue-0" java.lang.UnsupportedOperationException: getSource() not supported
	at androidx.compose.ui.graphics.PainterImage.getSource(DesktopImageConverters.desktop.kt:137)
	at java.desktop/javax.swing.plaf.metal.MetalUtils.getOceanDisabledButtonIcon(MetalUtils.java:404)
	at java.desktop/javax.swing.plaf.metal.MetalLookAndFeel.getDisabledIcon(MetalLookAndFeel.java:1672)
	at java.desktop/javax.swing.AbstractButton.getDisabledIcon(AbstractButton.java:638)
	at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI.paintIcon(BasicMenuItemUI.java:676)
	at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI.paintMenuItem(BasicMenuItemUI.java:660)
	at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI.paint(BasicMenuItemUI.java:621)
	at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI.update(BasicMenuItemUI.java:617)
	at java.desktop/javax.swing.JComponent.paintComponent(JComponent.java:797)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1074)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:907)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1083)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:907)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1083)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:907)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1083)
	at java.desktop/javax.swing.JLayeredPane.paint(JLayeredPane.java:586)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:907)
	at java.desktop/javax.swing.JComponent.paintToOffscreen(JComponent.java:5271)
	at java.desktop/javax.swing.RepaintManager$PaintManager.paintDoubleBufferedImpl(RepaintManager.java:1643)
	at java.desktop/javax.swing.RepaintManager$PaintManager.paintDoubleBuffered(RepaintManager.java:1618)
	at java.desktop/javax.swing.RepaintManager$PaintManager.paint(RepaintManager.java:1556)
	at java.desktop/javax.swing.RepaintManager.paint(RepaintManager.java:1323)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1060)
	at java.desktop/java.awt.GraphicsCallback$PaintCallback.run(GraphicsCallback.java:39)
	at java.desktop/sun.awt.SunGraphicsCallback.runOneComponent(SunGraphicsCallback.java:75)
	at java.desktop/sun.awt.SunGraphicsCallback.runComponents(SunGraphicsCallback.java:112)
	at java.desktop/java.awt.Container.paint(Container.java:2003)
	at java.desktop/java.awt.Window.paint(Window.java:3949)
	at java.desktop/javax.swing.RepaintManager$4.run(RepaintManager.java:876)
	at java.desktop/javax.swing.RepaintManager$4.run(RepaintManager.java:848)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:848)
	at java.desktop/javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:823)
	at java.desktop/javax.swing.RepaintManager.prePaintDirtyRegions(RepaintManager.java:772)
	at java.desktop/javax.swing.RepaintManager$ProcessingRunnable.run(RepaintManager.java:1884)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:316)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

Workaround is to rasterize vector before using it:
```
val icon = rememberVectorPainter(Icons.Default.Menu).rasterize()

@Composable
fun Painter.rasterize(): Painter {
    val density = LocalDensity.current
    val image = asAwtImage(density, LocalLayoutDirection.current) as MultiResolutionImage
    val bufferedImage = image.getResolutionVariant((16 * density.density).toDouble(), (16 * density.density).toDouble()) as BufferedImage
    return bufferedImage.asPainter()
}
```

**Comment by igordmn** [comment]
Should be fixed in 1.0.0-beta6-dev455.
